sunbird-rc-plugin/src/main/java/io/mosip/esignet/plugin/sunbirdrc/service/SunbirdRCAuthenticationService.java

```diff
@@ -110,7 +140,92 @@ public KycAuthResult doKycAuth(@NotBlank String relyingPartyId, @NotBlank String
     @Override
     public KycExchangeResult doKycExchange(String relyingPartyId, String clientId, KycExchangeDto kycExchangeDto)
             throws KycExchangeException {
-        throw new KycExchangeException(ErrorConstants.NOT_IMPLEMENTED);
+        String kycToken=kycExchangeDto.getKycToken();
+        Map<String,Object> responseRegistryMap;
+        try {
+            if (kycExchangeDto.getAcceptedClaims() == null) {
+                kycExchangeDto.setAcceptedClaims(new ArrayList<>());
+            }
+            if (kycExchangeDto.getAcceptedClaims() != null) {
```

Comment:
When you are checking the null and assigning empty list to acceptedClaims property. if condition in the line 149 will always be true. Is this "if" condition required?